boards/arm/olimex_stm32_p405/olimex_stm32_p405.dts

```diff
@@ -51,3 +51,10 @@
 &rtc {
 	status = "okay";
 };
+
+&can1 {
+	pinctrl-0 = <&can1_pins_b>;
```

Comment:
pinctrl is not used any more.

Reply:
@alexanderwachter  @galak @erwango is this a general thing to not use pinctrl anymore ? There are still CAN pins in the L4 and F0. For the rest it only seems uarts that use it.

And when that pin information isn't in DT anymore where is it? There are several pins that can be used as CAN pins (as you can see in the DT patch).  Somehow I find the DT usage of Zephyr inconsistent and confusing at times, on one side things like baudrate (a user runtime setting) are in there and on the other side things like the hardware pin description aren't ?

Reply:
Anyway I removed the pinctrl stuff to get this PR accepted.

Reply:
@lowlander, actually pinctrl has never been used. It has been pushed initially a bit early before we get the associated dt generation code code that could use it. It has been kept waiting for the dt generation code which has been much more delayed than initially though. Ideally it should be cleaned up, but will have to be re-instantiated sooner or later and now some people are using it of the tree.. So this is crap but a good lesson learnt.